test/e2e/user_group_limit/user_group_limit_test.go

```diff
@@ -909,6 +923,190 @@ var _ = ginkgo.Describe("UserGroupLimit", func() {
+		gomega.Ω(err).NotTo(HaveOccurred())
+		// Create a ClusterRole with necessary permissions
+		ginkgo.By("Creating ClusterRole...")
+		clusterRole := &rbacv1.ClusterRole{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "pod-creator-role",
+			},
+			Rules: []rbacv1.PolicyRule{
+				{
+					APIGroups: []string{""},
+					Resources: []string{"pods", "serviceaccounts", "test-user-sa"},
+					Verbs:     []string{"create", "get", "list", "watch", "delete"},
+				},
+			},
+		}
+		_, err = kClient.CreateClusterRole(clusterRole)
+		gomega.Ω(err).NotTo(HaveOccurred())
+		// Create a ClusterRoleBinding to bind the ClusterRole to the service account
+		ginkgo.By("Creating ClusterRoleBinding...")
+		clusterRoleBinding := &rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "pod-creator-role-binding",
+			},
+			RoleRef: rbacv1.RoleRef{
+				APIGroup: "rbac.authorization.k8s.io",
+				Kind:     "ClusterRole",
+				Name:     "pod-creator-role",
+			},
+			Subjects: []rbacv1.Subject{
+				{
+					Kind:      "ServiceAccount",
+					Name:      sa.Name,
+					Namespace: namespace,
+				},
+			},
+		}
+		_, err = kClient.CreateClusterRoleBinding(clusterRoleBinding.ObjectMeta.Name, clusterRoleBinding.RoleRef.Name, clusterRoleBinding.Subjects[0].Namespace, clusterRoleBinding.Subjects[0].Name)
+		gomega.Ω(err).NotTo(HaveOccurred())
+		// Create a Secret for the Service Account
+		ginkgo.By("Creating Secret for the Service Account...")
+		// create a object of v1.Secret
+		secret := &v1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: secretName,
+				Annotations: map[string]string{
+					"kubernetes.io/service-account.name": serviceAccountName,
+				},
+			},
+			Type: v1.SecretTypeServiceAccountToken,
+		}
+		time.Sleep(10 * time.Second)
+		_, err = kClient.CreateSecret(secret, namespace)
+		gomega.Ω(err).NotTo(HaveOccurred())
+		time.Sleep(10 * time.Second)
+		// Get the token value from the Secret
+		ginkgo.By("Getting the token value from the Secret...")
+		userTokenValue, err := kClient.GetSecretValue(namespace, secretName, "token")
+		gomega.Ω(err).NotTo(HaveOccurred())
+		// use deep copy not to hardcode the kubeconfig
+		config, err := kClient.GetKubeConfig()
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+		// Modify the TLS configuration to remove cert/key files for token-based auth
+		newConf := rest.CopyConfig(config) // copy existing config
+		// Use token-based authentication instead of client certificates
+		newConf.CAFile = ""
+		newConf.CertFile = ""
+		newConf.KeyFile = ""
+		// Set the Kubernetes API server URL
+		newConf.BearerToken = userTokenValue
+		newConf.Username = ""
+		newConf.Password = ""
+		kubeconfigPath := filepath.Join(os.TempDir(), "test-user-config")
+		err = k8s.WriteConfigToFile(newConf, kubeconfigPath)
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+		config, err = clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+		gomega.Ω(err).NotTo(HaveOccurred())
+		clientset, err = kubernetes.NewForConfig(config)
```

Comment:
Could we simplify this six line to:
```
clientset, err = kubernetes.NewForConfig(newConf)
```

Reply:
```suggestion
		clientset, err = kubernetes.NewForConfig(newConf)
```

Reply:
Tried the suggested solution but facing issue , It's not working with this test case.